CHANGELOG.md

```diff
@@ -1,4 +1,5 @@
 ## 1.0.4 (Unreleased)
+ * consul/connect: Enable setting `local_bind_address` field on connect upstreams [[GH-6248](https://github.com/hashicorp/nomad/issues/6248)]
```

Comment:
Like you already mentioned, we'll want this under the `1.1.0` header now, specifically under the `IMPROVEMENTS` sub header

Reply:
Updated changelog

Also added LocalBindAddress to interpolation function, and updated interpolation tests